src/machine/machinectl.c

```diff
@@ -1236,6 +1236,38 @@ static int process_forward(sd_event *event, PTYForward **forward, int master, PT
         return ret;
 }
 
+static int parse_machine_uid(const char *spec, const char **machine, char **uid) {
+        /*
+         * Whatever is specified in the spec takes priority over global arguments.
+         */
+        char *_uid;
+        const char *_machine = NULL;
+
+        if (spec) {
+                const char *at;
+
+                at = strchr(spec, '@');
+                if (at) {
+                        if (at == spec)
+                                return -EINVAL;
+
+                        _uid = strndup(spec, at - spec);
+                        _machine = at + 1;
+                } else {
+                        _uid = strdup(arg_uid ?: "root");
+                        _machine = spec;
+                }
+        } else
+                _uid = strdup(arg_uid ?: "root");
```

Comment:
so here's something I noticed: in the previous code we'd leave _uid as NULL if no name was specified, so that the systemd on the server side would fill in "root". With this change you'll fill this in on the client-side instead.

I figure the difference doesn't matter too much, but I wonder if it wouldn't be better to always leave this to the server side, maybe we want to alter this one day...